src/v/syschecks/syschecks.h

```diff
@@ -17,7 +17,10 @@
 #include <seastar/core/future.hh>
 #include <seastar/util/log.hh>
 
+#if !defined __arm__ && !defined __aarch64__
```

Comment:
i think it makes sense to do something like 

```
REDPANDA_IF_ARM(
#include<cputid.h>

)```

in `v/predefs.h` that imports boost/predef.h like @BenPope 

```
#define REDPANDA_IF_ARM(a, b) \
#If __arm...
a
#elseif
b
...

```

Reply:
or maybe since we have only one type `REDPANDA_IF_NOT_ARM()`

Reply:
besides Noah's comments, should this ☝️  be addressed now or can it be done later? (I ask because the PR was approved but these comments made me doubt whether I could merge)